hooks/hookexecution/executor.go

```diff
@@ -130,6 +132,35 @@ func (e *hookExecutor) ExecuteRawAuctionStage(requestBody []byte) ([]byte, *Reje
 	return payload, reject
 }
 
+func (e *hookExecutor) ExecuteProcessedAuctionStage(request *openrtb2.BidRequest) *RejectError {
+	plan := e.planBuilder.PlanForProcessedAuctionStage(e.endpoint, e.account)
+	if len(plan) == 0 {
+		return nil
+	}
+
+	handler := func(
+		ctx context.Context,
+		moduleCtx hookstage.ModuleInvocationContext,
+		hook hookstage.ProcessedAuctionRequest,
+		payload hookstage.ProcessedAuctionRequestPayload,
+	) (hookstage.HookResult[hookstage.ProcessedAuctionRequestPayload], error) {
+		return hook.HandleProcessedAuctionHook(ctx, moduleCtx, payload)
+	}
+
+	stageName := hooks.StageProcessedAuction.String()
+	executionCtx := e.newContext(stageName)
+	payload := hookstage.ProcessedAuctionRequestPayload{BidRequest: request}
+
+	outcome, _, contexts, reject := executeStage(executionCtx, plan, payload, handler, e.metricEngine)
```

Comment:
For my own understanding, this 2nd variable that we're ignoring here, I just wanted to ask why it's being ignored?

Reply:
Because we are passing the bidRequest as a pointer in the payload and any changes made during the apply of mutations should be reflected without the need to update the bidrequest passed to the `ExecuteProcessedAuctionStage` func. In the previous stages we passed the body as bytes and had to returned the updated payload after the execution.
If you think this could be misleading or may have other issues we can update with

```
outcome, payload, contexts, reject := executeStage(executionCtx, plan, payload, handler, e.metricEngine)
request = payload.BidRequest
```

Reply:
Thanks for the response, I think the code is fine as it is!